src/v/cluster/archival_metadata_stm.cc

```diff
@@ -311,45 +310,45 @@ ss::future<std::error_code> archival_metadata_stm::cleanup_metadata(
 
 ss::future<std::error_code> archival_metadata_stm::do_replicate_commands(
   model::record_batch batch,
-  ss::lowres_clock::time_point deadline,
   std::optional<std::reference_wrapper<ss::abort_source>> as) {
+    auto current_term = _insync_term;
     auto fut = _raft->replicate(
-      _insync_term,
+      current_term,
       model::make_memory_record_batch_reader(std::move(batch)),
       raft::replicate_options{raft::consistency_level::quorum_ack});
 
+    // Run with an abort source so shutdown doesn't have to wait a full
+    // replication timeout to proceed.
     if (as) {
-        fut = ssx::with_timeout_abortable(std::move(fut), deadline, *as);
-    } else {
-        fut = ss::with_timeout(deadline, std::move(fut));
-    }
-
-    result<raft::replicate_result> result{{}};
-    try {
-        result = co_await std::move(fut);
-    } catch (const ss::timed_out_error&) {
-        result = errc::timeout;
+        fut = ssx::with_timeout_abortable(
+          std::move(fut), model::no_timeout, *as);
```

Comment:
Why do we need `with_timeout` here? only to add an abort source?
If this is the case maybe it's OK to remove this because this abort source will only be used to stop `ntp_archiver` when redpanda stops or raft group is no longer a leader. In both cases the future returned by `replicate` call will transition to failed state.

Reply:
Also, it's probably OK to keep this as a precaution.

Reply:
Yeah, I left this in mostly as a precaution since it seems like it was added originally to avoid flakiness on shutdown.